testing/tests/test_runtime.py

```diff
@@ -117,3 +117,19 @@ def test_env_clean_on_charm_error():
 
     # Ensure that the Juju environment didn't leak into the outside one.
     assert os.getenv("JUJU_REMOTE_APP", None) is None
+
+
+def test_juju_version_is_set_in_environ():
+    version = "2.9"
+
+    class MyCharm(ops.CharmBase):
+        def __init__(self, framework: ops.Framework):
+            super().__init__(framework)
+            framework.observe(self.on.start, self._on_start)
+
+        def _on_start(self, _: ops.StartEvent):
+            juju_version = ops.JujuVersion.from_environ()
+            assert juju_version == version
+
+    ctx = Context(MyCharm, meta={"name": "foo"}, juju_version=version)
```

Comment:
Minor: I think this would read clearer:

```suggestion
            assert ops.JujuVersion.from_environ() == "2.9"

    ctx = Context(MyCharm, meta={"name": "foo"}, juju_version="2.9")
```

Reply:
I'm happy to collapse the 131 and 132 lines into a single line. I'd rather not repeat the `2.9` string, though.